src/java/org/apache/cassandra/io/compress/CompressedSequentialWriter.java

```diff
@@ -109,7 +109,7 @@ public CompressedSequentialWriter(File file,
                             .bufferType(parameters.getSstableCompressor().preferredBufferType())
                             .finishOnClose(option.finishOnClose())
                             .build());
-        ICompressor compressor = parameters.getSstableCompressor();
```

Comment:
alignement, plus we should most probably not use it like this. We should still use `getSstableCompressor()` as it was before. In there, we decide what compressor we want to use. If plugin compressor is not available, then `decoratedSstableCompressor` will be null. The we will do

    public ICompressor getSstableCompressor()
    {
        return pluginCompressor != null ? pluginCompressor : sstableCompressor;
    }

If we do it like that, then we most probably realize that we do not need `CompressorDecorator` and `AbstractCompressorDecorator` at all. Because the plugin compressor would just implement `ICompressor` already.

I do not see why we would need to fall back to base compressor in case plugin compressor is unable to un/compress.